File: 01_Java/Exercise/2주차/Labs/Lab9-1/Starter/A.cs
using System;

class A {
    public virtual a() {
        Console.WriteLine("A");
    }
}

class B : A {
    public override a() {
        Console.WriteLine("B");
    }
}

class C : B {
    public new virtual a() {
        Console.WriteLine("C");
    }
}

class D : C {
    public override a() {
        Console.WriteLine("D");
    }
}

class Test {
    public static void main(string[] args) {
        D d = new D();
        C c  = d;
        B b  = c;
        A a = b;
    }
}

File: 01_Java/Exercise/2주차/Labs/Lab9-1/Starter/A.java
class A {
    public static void a() {
        System.out.println("A");
    }
}

class B extends A {
    public static void a() {
        System.out.println("B");
    }
}

class C extends B {
    public static void a() {
        System.out.println("C");
    }
}

class D extends C {
    public static void a() {
        System.out.println("D");
    }
}

class Test {
    public static void main(String[] args) {
        D d = new D();
        C c  = d;
        B b  = c;
        A a = b;
    }
}